airtime_mvc/application/controllers/LibraryController.php

```diff
@@ -308,6 +311,42 @@ public function duplicateAction(){
         $newPl->setName(sprintf(_("Copy of %s"), $originalPl->getName()));
     }
 
+    // duplicate smartblock
+    public function duplicateBlockAction(){
+        Logging::info("duplicate smartblock functionality not yet implemented");
+        $params = $this->getRequest()->getParams();
+        $id = $params['id'];
+        Logging::info($params);
+
+        $originalBl = new Application_Model_Block($id);
+        $newBl = new Application_Model_Block();
+        $newBl->setCreator(Application_Model_User::getCurrentUser()->getId());
+        $newBl->setDescription($originalBl->getDescription());
+        if ($originalBl->isStatic()) {
+            $newBl->saveType('static');
+        }
+        else {
+            $newBl->saveType('dynamic');
+        }
+        // the issue here is that the format that getCriteria provides is different from the format the saveCriteria
+        // expects due to the useage of startForm. So we either need to write new code that simply copies the database
+        // or figure out a way to instantiate a form inside of here and save it without modifying it.
+        //$newBlForm = new Application_Form_SmartBlockCriteria;
```

Comment:
I think we would need to wrap `Application_Form_SmartBlockCriteria` in a proper form since it is based on `Zend_Form_SubForm` to be able to use the form.